lib/iris/_concatenate.py

```diff
@@ -991,6 +1019,14 @@ def register(
             if existing_order == _CONSTANT and this_order != _CONSTANT:
                 self._coord_signature.dim_order[dim_ind] = this_order
 
+        if mismatch_error_msg:
+            if not match and error_on_mismatch:
+                raise iris.exceptions.ConcatenateError([mismatch_error_msg])
+            elif not match:
+                warnings.warn(
+                    mismatch_error_msg, category=iris.warnings.IrisUserWarning
+                )
```

Comment:
Need test coverage for this block. I know the tests are still using `unittest`, but it should be possible to sneak `pytest.mark.parametrize` and `pytest.warns()` in there without it crashing.